sdk/js/src/service/gateways.js

```diff
@@ -17,13 +17,61 @@ import Marshaler from '../util/marshaler'
 class Gateways {
   constructor (api, { defaultUserId, stackConfig, proxy = true }) {
     this._api = api
+    this._defaultUserId = defaultUserId
   }
 
-  async getAll (params) {
-    const response = await this._api.GatewayRegistry.List(undefined, params)
+  // Retrieval
+
+  async getAll (params, selector) {
+    const response = await this._api.GatewayRegistry.List(undefined, {
+      ...params,
+      ...Marshaler.selectorToFieldMask(selector),
+    })
 
     return Marshaler.unwrapGateways(response)
   }
+
+  async getById (id, selector) {
+    const fieldMask = Marshaler.selectorToFieldMask(selector)
+    const response = await this._api.GatewayRegistry.Get({
+      routeParams: { 'gateway_ids.gateway_id': id },
+    }, fieldMask)
+
+    return Marshaler.unwrapGateway(response)
+  }
+
+  // Update
+
+  async updateById (id, patch, mask = Marshaler.fieldMaskFromPatch(patch)) {
+    const response = await this._api.GatewayRegistry.Update({
+      routeParams: { 'gateway.ids.gateway_id': id },
+    },
+    {
+      gateway: patch,
+      field_mask: Marshaler.fieldMask(mask),
+    })
+
+    return Marshaler.unwrapGateway(response)
+  }
+
+  // Create
+
+  async create (userId = this._defaultUserId, gateway) {
+    const response = await this._api.GatewayRegistry.Create({
+      routeParams: { 'collaborator.user_ids.user_id': userId },
+    },
+    { gateway })
+
+    return Marshaler.unwrapGateway(response)
+  }
+
+  // Delete
+
+  async deleteById (id) {
```

Comment:
I think it's better if we also unwrap the results here using `Marshaler.payloadSingleResponse()`. Same thing with `Applications.delete()`.